openapi-core/src/main/java/org/openapi4j/core/validation/ValidationResults.java

```diff
@@ -166,50 +163,103 @@ public String toString() {
   public static class ValidationItem extends ValidationResult implements Serializable {
     private static final long serialVersionUID = 7905122048950251207L;
 
+    private static final String SCHEMA_CRUMB_START = "<";
+    private static final String SCHEMA_CRUMB_END = ">";
     private static final String DOT = ".";
-    private static final String SEMI_COLON = " : ";
+    private static final String SEMI_COLON = ": ";
 
-    private final String crumbs;
+    private final Collection<CrumbInfo> crumbs;
 
-    ValidationItem(ValidationResult result, Collection<String> crumbs, Object... msgArgs) {
+    ValidationItem(ValidationResult result, Collection<CrumbInfo> crumbs, Object... msgArgs) {
       this(result, crumbs, null, msgArgs);
     }
 
-    ValidationItem(ValidationResult result, Collection<String> crumbs, String crumb, Object... msgArgs) {
+    ValidationItem(ValidationResult result, Collection<CrumbInfo> crumbs, CrumbInfo crumbInfo, Object... msgArgs) {
       super(
         result.severity(),
         result.code(),
         (msgArgs != null) ? String.format(result.message(), msgArgs) : result.message());
 
-      this.crumbs = joinCrumbs(crumbs, crumb);
+      this.crumbs = new ArrayList<>(crumbs);
+      if (crumbInfo != null) {
+        this.crumbs.add(crumbInfo);
+      }
+    }
+
+    /**
+     * Compile and get data path.
+     * Warning, the compilation occurs at each call.
+     *
+     * @return The data path.
+     */
+    public String dataCrumbs() {
+      return joinCrumbs(crumbs, true);
     }
 
-    public String crumbs() {
-      return crumbs;
+    /**
+     * Compile and get schema path definition.
+     * Warning, the compilation occurs at each call.
+     *
+     * @return The schema path.
+     */
+    public String schemaCrumbs() {
+      return joinCrumbs(crumbs, false);
+    }
+
+    private String joinCrumbs(Collection<CrumbInfo> crumbs, boolean dataCrumbs) {
+      if (dataCrumbs) {
+        return crumbs
+          .stream()
+          .filter(crumbInfo -> !crumbInfo.isSchemaCrumb() && crumbInfo.crumb() != null)
+          .map(CrumbInfo::crumb)
+          .collect(Collectors.joining(DOT));
+      }
+
+      StringJoiner joiner = new StringJoiner(DOT);
+      for (CrumbInfo crumb : crumbs) {
+        if (crumb.crumb() == null) continue;
+
+        if (crumb.isSchemaCrumb()) {
+          joiner.add(SCHEMA_CRUMB_START + crumb.crumb() + SCHEMA_CRUMB_END);
+        } else {
+          joiner.add(crumb.crumb());
+        }
+      }
+
+      return joiner.toString();
     }
 
     @Override
     public String toString() {
```

Comment:
We probably don't need to output both data crumbs and schema crumbs as part of 'toString', given that schemaCrumbs is effectively a superset of datacrumbs

Reply:
toString() is overridden for debugging purpose.